Firestore/Protos/CMakeLists.txt

```diff
@@ -98,63 +98,64 @@ target_include_directories(
   ${FIREBASE_SOURCE_DIR}/Firestore/Protos/cpp
 )
 
-set(NANOPB_GENERATOR_DIR ${FIREBASE_BINARY_DIR}/external/src/nanopb/generator)
 
 set(PROTOBUF_DIR ${FIREBASE_BINARY_DIR}/external/src/grpc/third_party/protobuf)
-set(DESCRIPTOR_PROTO_SRC
+set(NANOPB_DIR ${FIREBASE_BINARY_DIR}/external/src/nanopb)
+#set(OUTPUT_DIR ${CMAKE_CURRENT_BINARY_DIR}/generated_protos)
+set(NANOPB_OUTPUT_DIR ${CMAKE_CURRENT_SOURCE_DIR})
+
+set(
+  PROTOBUF_PROTO
   ${PROTOBUF_DIR}/src/google/protobuf/descriptor.proto
+  ${NANOPB_DIR}/generator/proto/nanopb.proto
+  ${NANOPB_DIR}/generator/proto/plugin.proto
 )
-set(DESCRIPTOR_PYTHON_SRC
+
+set(
+  PROTOBUF_PYTHON
   ${PROTOBUF_DIR}/python/google/protobuf/descriptor_pb2.py
+  ${PROTOBUF_DIR}/python/proto/nanopb_pb2.py
+  ${PROTOBUF_DIR}/python/proto/plugin_pb2.py
+)
+
+set(
+  PROTO_INCLUDES
+  -I${CMAKE_CURRENT_SOURCE_DIR}/protos
+  -I${NANOPB_DIR}/generator
+  -I${PROTOBUF_DIR}/src
 )
+
 add_custom_command(
-  OUTPUT ${DESCRIPTOR_PYTHON_SRC}
+  COMMENT "Generating protoc python plugins"
+  OUTPUT
+    ${PROTOBUF_PYTHON}
   COMMAND
-    protoc
-      -I${PROTOBUF_DIR}/src
+    $<TARGET_FILE:protoc>
```

Comment:
Fine, but you can simplify this to just `protoc`. cmake knows to look for an internal binary named `protoc` before searching $PATH. Or in other words, cmake interprets `protoc` as `$<TARGET_FILE:protoc>` when it's the first "word" in a `COMMAND`. (This is a cmake 2.something feature where something is much less than our current target version.)

(OTOH, you *do* need it if it's not the first argument since cmake doesn't realize it's supposed to be a path to an executable in that case.)

Reply:
Done.